internal/bgp/frr/config.go

```diff
@@ -64,14 +67,41 @@ router bgp {{.MyASN}} view {{.MyASN}}
     network {{.Prefix}}
   exit-address-family
 {{- end}}
-{{end}}
-{{end}}
-`
+{{end -}}
+{{end }}
+{{- if gt (len .BFDProfiles) 0}}
+bfd
+{{- range .BFDProfiles }}
+  profile {{.Name}}
+    {{ if .ReceiveInterval -}}
+    receive-interval {{.ReceiveInterval}}
+    {{end -}}
+    {{ if .TransmitInterval -}}
+    transmit-interval {{.TransmitInterval}}
+    {{end -}}
+    {{ if .DetectMultiplier -}}
+    detect-multiplier {{.DetectMultiplier}}
+    {{end -}}
+    {{ if .EchoMode -}}
+    echo-mode
+    {{end -}}
+    {{ if .EchoInterval -}}
+    echo-interval {{.EchoInterval}}
```

Comment:
I don't think the echo interval of the BFD in frr is configured this way.
In their docs it is splits to:
echo receive-interval <disabled|(10-60000)>
echo transmit-interval (10-60000)

Reply:
That part changed between 7.5 (which we are pinning now) and latest.

Reply:
maybe I am missing something in the config there should be a matching peer section in the bfd config ? the above configure only the profile ?

Reply:
You can associate a neighbor to a bfd profile, which is what happens in https://github.com/metallb/metallb/pull/967/files#diff-1d9a33c302812585b1b2d1897f84126492129d1f9415786a73dcfa26603686c9R51

Reply:
Documented here https://docs.frrouting.org/en/latest/bfd.html#bgp-bfd-configuration

Reply:
We are not configuring bfd peers explicitly because we are leveraging the bgp's bfd option described here https://docs.frrouting.org/en/latest/bfd.html#bgp-bfd-configuration

If you have a bfd profile defined, it's totally fine to add the option to the neighbour and that neighbor will be backed up by a bfd session. So the configuration is more compact without the need of adding the peer explicitly.
This is what we do here https://github.com/metallb/metallb/pull/967/commits/694ed3476a26297dfca1db98f858be801104f0ca#diff-1d9a33c302812585b1b2d1897f84126492129d1f9415786a73dcfa26603686c9R51

Reply:
> do u know how this will work with view bgp do u have one bfd process for every bgp instance or its shared and in that case u pick peers from both ?

I don't know, I guess we'll figure it out if / when we re-enable views.